test/test.js

```diff
@@ -0,0 +1,608 @@
+let fs = require('fs')
+let assert = require('assert')
+let {
+    createRotatable
+} = require('../index')
+
+function nextWrite(data, streams, callBack, expectedResults) {
+
+    if (!expectedResults) {
+        expectedResults = {}
+        streams.forEach(stream => {
+            expectedResults[stream.name] = {
+                writeCount: 0,
+                written: 0
+            }
+        })
+    }
+
+    if (!data.length) {
+        streams.forEach(stream => {
+            stream.instance.end()
+        })
+        callBack(expectedResults)
+    }
+
+    let d = data.pop()
+    let writes = 0
+
+    streams.forEach(stream => {
+        let name = stream.name
+        d.w = name
+        let body = JSON.stringify(d) + '\n'
+        stream.instance.write(body, function () {
+            expectedResults[name].written += body.length
+            if (++writes === streams.length) {
+                ++expectedResults[name].writeCount
+                nextWrite(data, streams, callBack, expectedResults)
+            }
+        })
+    })
+}
+
+function cleanup() {
+    fs.readdirSync('.')
+        .filter(function (path) {
+            return path.indexOf('test.log') === 0;
+        })
+        .forEach(function (path) {
+            fs.unlinkSync(path);
+        })
+}
+
+function readFiles() {
+    return fs.readdirSync('.')
+        .filter(function (path) {
+            return path.indexOf('test.log') === 0;
+        })
+        .reduce(function (files, path) {
+            files.push({
+                path: path,
+                body: fs.readFileSync(path).toString(),
+                stats: fs.statSync(path)
+            })
+            return files;
+        }, [])
+}
+
+describe('rotatable tests', function () {
+
+    describe('prior version tests', function () {
+        let results = {
+            written: 0,
+            rotates: []
+        }
+
+        let data = [{
```

Comment:
Whoa, that's a very weird structure. The canonical form of list of objects is:

```js
let data = [
    {i: 1},
    {i: 2},
    // ...
]``

Reply:
Also relevant for the other cases in this file.

Reply:
tnx, addressed